`cagent run <()` panics

cagent run <()
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0xb8 pc=0x101500344]

goroutine 1 [running]:
github.com/docker/cagent/pkg/agent.(*Agent).MaxIterations(...)
	github.com/docker/cagent/pkg/agent/agent.go:81
github.com/docker/cagent/cmd/root.doRunCommand({0x101e19d28, 0x102c7a040}, {0x14000813a40, 0x1, 0x1}, 0x0)
	github.com/docker/cagent/cmd/root/run.go:247 +0x944
github.com/docker/cagent/cmd/root.runCommand(0x140001e9808, {0x14000813a40, 0x1, 0x1})
	github.com/docker/cagent/cmd/root/run.go:86 +0x64
github.com/spf13/cobra.(*Command).execute(0x140001e9808, {0x14000813a10, 0x1, 0x1})
	github.com/spf13/cobra@v1.10.1/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x140001e9208)
	github.com/spf13/cobra@v1.10.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.10.1/command.go:1071
github.com/docker/cagent/cmd/root.Execute()
	github.com/docker/cagent/cmd/root/root.go:151 +0x130
github.com/docker/cagent/cmd/root.Run(...)
	github.com/docker/cagent/cmd/root/root.go:129
main.main()
	./main.go:6 +0x20

--- a/pkg/runtime/runtime.go
+++ b/pkg/runtime/runtime.go
@@ -124,6 +124,18 @@ func New(agents *team.Team, opts ...Opt) (Runtime, error) {
 		opt(r)
 	}
 
+	// Validate that we have at least one agent and that the current agent exists
+	if r.team == nil || r.team.Size() == 0 {
+		return nil, fmt.Errorf("no agents loaded; ensure your agent configuration defines at least one agent")
+	}
+	if r.team.Agent(r.currentAgent) == nil {
+		names := strings.Join(r.team.AgentNames(), ", ")
+		if names == "" {
+			names = "(none)"
+		}
+		return nil, fmt.Errorf("agent %q not found in team; available agents: %s", r.currentAgent, names)
+	}
+
 	slog.Debug("Creating new runtime", "agent", r.currentAgent, "available_agents", agents.Size())
 
 	return r, nil

--- a/pkg/runtime/runtime_test.go
+++ b/pkg/runtime/runtime_test.go
@@ -417,3 +417,23 @@ func TestSessionWithoutUserMessage(t *testing.T) {
 	require.True(t, hasEventType(t, events, &StreamStoppedEvent{}), "Expected StreamStoppedEvent")
 	require.False(t, hasEventType(t, events, &UserMessageEvent{}), "Should not have UserMessageEvent when SendUserMessage is false")
 }
+
+func TestNewRuntime_NoAgentsError(t *testing.T) {
+	tm := team.New()
+
+	_, err := New(tm, WithModelStore(mockModelStore{}))
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "no agents loaded")
+}
+
+func TestNewRuntime_InvalidCurrentAgentError(t *testing.T) {
+	// Create a team with a single agent named "root"
+	root := agent.New("root", "You are a test agent")
+	tm := team.New(team.WithAgents(root))
+
+	// Ask for a non-existent current agent
+	_, err := New(tm, WithCurrentAgent("other"), WithModelStore(mockModelStore{}))
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "agent \"other\" not found")
+	require.Contains(t, err.Error(), "root") // available agents listed in error
+}

